libs/components/src/packing-list/label/ItemLabelForm.tsx

```diff
@@ -0,0 +1,203 @@
+
+      // Adding new label to Firestore will generate a new ID, so we need to re-fetch labels and update Redux
+      await refetchLabels(firebase, auth, dispatch)
+
+      trackEvent('User Label Created', {
+        label: values.labelText,
+        color: values.labelColor
+      })
+    } catch (error) {
+      trackEvent('User Label Create Failure', {
+        label: values.labelText,
+        color: values.labelColor,
+        error,
+      })
+      toast.error('Failed to add label, please try again')
+    } finally {
+      // Return user to the select list
+      setSubmitting(false)
+      toggleListHandler()
+    }
+  }
+
+  const handleUpdate = async (values: any, {resetForm, setSubmitting}: any) => {
+    resetForm({})
+    setSubmitting(true)
+
+    // Handle form submission
+    try {
+      if (!label) throw new Error('Label not found')
+
+      await firebase
+        .firestore()
+        .collection('users')
+        .doc(auth.uid)
+        .collection('labels')
+        .doc(label.id)
+        .update({
+          text: values.labelText,
+          color: values.labelColor
+        })
+
+      await refetchLabels(firebase, auth, dispatch)
+
+      trackEvent('User Label Updated', {
+        label: values.labelText,
+        color: values.labelColor
+      })
+    } catch (error) {
+      trackEvent('User Label Update Failure', {
+        label: values.labelText,
+        color: values.labelColor[0],
+        error,
+      })
+      toast.error('Failed to update label, please try again')
+    } finally {
+      // Return user to the select list
+      setSubmitting(false)
+      toggleListHandler()
+    }
+  }
+
+  return (
+    <Formik
+      initialValues={{
+        labelText: labelText,
+        labelColor: labelColor
+      }}
+      onSubmit={label ? handleUpdate : handleSubmit}
+      enableReinitialize={true}
+    >
+      {({ handleSubmit, isSubmitting, initialValues }) => (
+        <Form onSubmit={handleSubmit} onChange={handleChange}>
+          <FormWrapper>
+            <ItemLabelPreview color={labelColor} text={labelText} />
+            <ColorPickerInput disabled={isSubmitting} setColor={setLabelColor} initialValue={initialValues.labelColor as LabelColorName} />
+            <Field
+              as={Input}
+              type="text"
+              name={`labelText`}
+              initialValue={initialValues.labelText}
+              label="Label Name"
+              hiddenLabel
+              disabled={isSubmitting}
+              maxLength={32}
+            />
+            <SubmitButton type={'submit'}>
+              <FaPlus />
+              <CreateText>{ label?.id ? 'Update' : 'Create'}</CreateText>
+            </SubmitButton>
+          </FormWrapper>
+        </Form>
+      )}
+    </Formik>
+  )
+}
+
+/**
+ * Update redux store with the latest labels from Firestore
+ * @param firebase
+ * @param auth
+ * @param dispatch
+ */
+const refetchLabels = async (firebase: any, auth: any, dispatch: any) => {
```

Comment:
i think if you are using the `useFirestoreConnect()` hook to fetch the labels initially, and then you update the labels elsewhere, the hook will handle any updates (via subscription) and you shouldn't need to refetch. All you need to do is subscribe to the redux store and get the labels out that way